### PR TITLE
issuegen: do not use staged file, and fix bug in udev add/remove invocation

### PR DESCRIPTION
**Pull request by rfairley**
Fixes: #35 

No functional changes, contains two bugfixes. Please see commit messages for additional detail.

### CODE REVIEW

**Comment by dustymabe** [comment]
I don't know if we are solving the problem here. It still seems like we could have two processes writing to `$generated` at the same time, right? I think you'd either want to keep the old "staged" model and just use a new temporary file for each invocation OR aggregate all the content you want to write to the file in a bash variable and then write it all out in one command.

**Comment by rfairley** [comment]
> I don't know if we are solving the problem here. It still seems like we could have two processes writing to $generated at the same time, right? 

Right - the output could become corrupted still by the two processes writing to the generated file at the same time. This change just avoids having the service fail from the error missing the common `staged` file - but agreed it is not really solving the problem, it would be better to ensure that the output to the file is always correct.

> I think you'd either want to keep the old "staged" model and just use a new temporary file for each invocation OR aggregate all the content you want to write to the file in a bash variable and then write it all out in one command.

A new temporary file for each SGTM! Will update.

**Comment by dustymabe** [comment]
> A new temporary file for each SGTM! Will update.

it would be pretty easy to just use variables and not have to use a staged file I think. Something like:

```
generated_string=$(cat /etc/${ISSUE_DIR_PRIVATE}/* 2>/dev/null)
generated_string+=$(cat /run/${ISSUE_DIR_PRIVATE}/* 2>/dev/null)
generated_string+=$(cat /usr/lib/${ISSUE_DIR_PRIVATE}/* 2>/dev/null)

echo "${generated_string}" > "${generated}"
```
```

**Comment by rfairley** [comment]
Updated now - works well with variables.

Also rebased #40 which follows up.

**Comment by rfairley** [comment]
^ Re-pushed for a minor correction in the commit message.

File: usr/libexec/console-login-helper-messages/issuegen [hunk 1]
```diff
@@ -45,20 +36,20 @@ case "$1" in
...
+#       see https://github.com/karelzak/util-linux/commit/1fc82a1360305f696dc1be6105c9c56a9ea03f52#commitcomment-27949895
+#       until then, $GENERATED_ISSUE writes to the privately scoped directory (not in /run/issue.d)
+#
+# Pick 40 as an index as other files can order around it easily.
+generated="/run/${PKG_NAME}/40_${PKG_NAME}.issue"
+generated_string=''
+# Hack around files potentially not existing in the below paths with `|| true`.
+generated_string+=$(cat /etc/${ISSUE_DIR_PRIVATE}/* 2>/dev/null || true)
+generated_string+=$(cat /run/${ISSUE_DIR_PRIVATE}/* 2>/dev/null || true)
+generated_string+=$(cat /usr/lib/${ISSUE_DIR_PRIVATE}/* 2>/dev/null || true)
```
Comment (dustymabe) [inline usr/libexec/console-login-helper-messages/issuegen:53]: hmm. I hadn't thought about skipping the directory check, but I think this still works 👍

File: usr/libexec/console-login-helper-messages/issuegen [hunk 2]
```diff
@@ -45,20 +36,20 @@ case "$1" in
...
-if [[ -d "/usr/lib/${ISSUE_DIR_PRIVATE}" ]]; then
-	cat /usr/lib/${ISSUE_DIR_PRIVATE}/* 2>/dev/null >> "${staged}" || true
-fi
+# TODO: it would be nice to have /run/issue.d be an official directory,
+#       see https://github.com/karelzak/util-linux/commit/1fc82a1360305f696dc1be6105c9c56a9ea03f52#commitcomment-27949895
+#       until then, $GENERATED_ISSUE writes to the privately scoped directory (not in /run/issue.d)
+#
+# Pick 40 as an index as other files can order around it easily.
+generated="/run/${PKG_NAME}/40_${PKG_NAME}.issue"
+generated_string=''
```
Comment (dustymabe) [inline usr/libexec/console-login-helper-messages/issuegen:49]: slight nit: if we're going to use `generated_string` let's maybe use `generated_file` for the other variable

File: usr/libexec/console-login-helper-messages/issuegen (hunk 1 above)
Comment (rfairley) [reply usr/libexec/console-login-helper-messages/issuegen:53]: :+1: those directories are created upon installation of the RPM - but even if they do exist, `cat` will still give an error (tried it without `|| true`). Was also in the original script - https://github.com/coreos/init/blob/8b011b2686eaa963ac00923527aeb342116e55b2/scripts/motdgen#L18.

File: usr/libexec/console-login-helper-messages/issuegen (hunk 2 above)
Comment (rfairley) [reply usr/libexec/console-login-helper-messages/issuegen]: Updated, thanks!

**Comment by rfairley** [comment]
Will merge now - suggested fixups were added after approval.

**Pull request merged**